cmd/tail_test.go

```diff
@@ -0,0 +1,121 @@
+package cmd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLineWriter(t *testing.T) {
+	// Setup a writer with test message and output channel. Also mark time before any messages.
+	const msg = "a complete line"
+	out := make(chan line, 1)
+	lw := lineWriter{name: "mine", out: out}
+
+	// Write a message, mark after it's written but before we finish to verify Write produced it.
+	start := time.Now()
+	n, err := lw.Write([]byte(msg + "\n"))
+	assert.NoError(t, err)
+	assert.Equal(t, len(msg)+1, n)
+	mark := time.Now()
+	lw.Finish()
+	ln, ok := <-out
+	assert.True(t, ok)
+	assert.NoError(t, ln.Err)
+	assert.Equal(t, "mine", ln.source)
+	assert.Equal(t, msg, ln.Text)
+	assert.True(t, start.Before(ln.Time))
+	assert.True(t, mark.After(ln.Time))
+
+	start = time.Now()
+	n, err = lw.Write([]byte(msg + "\r"))
+	assert.NoError(t, err)
+	assert.Equal(t, len(msg)+1, n)
+	mark = time.Now()
+	lw.Finish()
+	ln, ok = <-out
+	assert.True(t, ok)
+	assert.NoError(t, ln.Err)
+	assert.Equal(t, "mine", ln.source)
+	assert.Equal(t, msg, ln.Text)
+	assert.True(t, start.Before(ln.Time))
+	assert.True(t, mark.After(ln.Time))
+
+	// Classic Windows endings, e.g. CRLF
+	start = time.Now()
+	n, err = lw.Write([]byte(msg + "\r\n"))
+	assert.NoError(t, err)
+	assert.Equal(t, len(msg)+2, n)
+	mark = time.Now()
+	lw.Finish()
+	ln, ok = <-out
+	assert.True(t, ok)
+	assert.NoError(t, ln.Err)
+	assert.Equal(t, "mine", ln.source)
+	assert.Equal(t, msg, ln.Text)
+	assert.True(t, start.Before(ln.Time))
+	assert.True(t, mark.After(ln.Time))
+
+	// Test message split over multiple writes
+	start = time.Now()
+	split := len(msg) / 2
+	n, err = lw.Write([]byte(msg[:split]))
+	assert.NoError(t, err)
+	assert.Equal(t, split, n)
+	n, err = lw.Write([]byte(msg[split:]))
+	assert.NoError(t, err)
+	assert.Equal(t, len(msg)-split, n)
+	n, err = lw.Write([]byte{'\r', '\n'})
+	assert.NoError(t, err)
+	assert.Equal(t, 2, n)
+	mark = time.Now()
+	lw.Finish()
+	ln, ok = <-out
+	assert.True(t, ok)
```

Comment:
There's a lot of copy-paste assertions here. Could these be more higher-level, e.g. `assertLine` or something? Not a PR blocker, but if it's easy to do then it would make the tests readable.